presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -94,7 +100,32 @@ public OrcReader(
                 orcDataSource,
                 orcEncoding,
                 orcFileTailSource,
-                stripeMetadataSource,
+                StripeMetadataSourceFactory.of(stripeMetadataSource),
```

Comment:
I kept the old constructor and added a new constructor with the StripeMetadataSourceFactory. I kept the old constructor to avoid changing multiple (10+) files not really related to this change. Depending on the reviewer's feedback I can keep it as is if it's not a big deal, or I cab make changes in all other places in a separate house-cleaning PR.